### PR TITLE
add support for ES algorithms

### PR DESCRIPTION
**Pull request by swaeberle**
See #117

### CODE REVIEW

**Review by JoshMcCullough — CHANGES_REQUESTED** [review]
Thanks for this. Can you please add tests matching those for RS*?

**Comment by swaeberle** [comment]
> Thanks for this. Can you please add tests matching those for RS*?

I'll have a look on this, too!

**Comment by swaeberle** [comment]
Unfortunately, I was not able to run the tests locally due to missing authentication with docker.io.
```
Building test runner using port 8000...
[+] Building 1.0s (3/3) FINISHED                                                                                                                                                                                                      docker:default
 => [nginx internal] load build definition from Dockerfile-test-nginx                                                                                                                                                                           0.0s
 => => transferring dockerfile: 601B                                                                                                                                                                                                            0.0s
 => [nginx internal] load .dockerignore                                                                                                                                                                                                         0.0s
 => => transferring context: 2B                                                                                                                                                                                                                 0.0s
 => ERROR [nginx internal] load metadata for docker.io/teslagov/jwt-nginx:1.24.0                                                                                                                                                                0.9s
------
 > [nginx internal] load metadata for docker.io/teslagov/jwt-nginx:1.24.0:
------
failed to solve: teslagov/jwt-nginx:1.24.0: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
```

File: test/Dockerfile-test-nginx [hunk 1]
```diff
@@ -9,4 +9,7 @@ RUN echo "Config Hash: ${CONFIG_HASH}"
 COPY /docker-entrypoint.d/* /docker-entrypoint.d/
 COPY /etc/nginx/conf.d/test.conf /etc/nginx/conf.d/test.conf
 COPY /etc/nginx/rsa_key_2048-pub.pem /etc/nginx/rsa-key.conf
+COPY /etc/nginx/ec_key_256-pub.pem /etc/nginx/ec-256-key.conf
+COPY /etc/nginx/ec_key_384-pub.pem /etc/nginx/ec-384-key.conf
+COPY /etc/nginx/ec_key_521-pub.pem /etc/nginx/ec-521-key.conf
```
Comment (JoshMcCullough) [inline test/Dockerfile-test-nginx:14]: `521` should be `512`.

File: test/ec_key_521.pem
Comment (JoshMcCullough) [inline test/ec_key_521.pem:1]: Should be `512`.

File: test/etc/nginx/ec_key_521-pub.pem
Comment (JoshMcCullough) [inline test/etc/nginx/ec_key_521-pub.pem:1]: Should be `512`.

**Review by JoshMcCullough — CHANGES_REQUESTED** [review]
Sorry for the delay. It's close -- make sure all instances of ES 512 are `512` and not `521`.

**Comment by swaeberle** [comment]
> Sorry for the delay. It's close -- make sure all instances of ES 512 are `512` and not `521`.

In fact the naming is no mixup but intended.

I used the elliptic curve `NIST P-521` to create the keys with `-521` suffix. Only later the key is used in combination with `SHA-512` to form JWT algorithm `ES512` (as defined by [RFC 7518 section 3.4](https://www.rfc-editor.org/rfc/rfc7518#section-3.4)).

So the number originates from the named elliptic curve, not the hash function.

Though for the other JWT algorithms, named curve and hash function are equivalent.

**Comment by beeing** [comment]
Hi, any resolution on this? I'm interested to use it with this too. Thanks in advance!

**Comment by swaeberle** [comment]
@JoshMcCullough Are you fine with keeping the test keys file names or do you insist in renaming them?

**Comment by JoshMcCullough** [comment]
> @JoshMcCullough Are you fine with keeping the test keys' file names or do you insist in renaming them?

Thank you for clarifying -- my mistake.

**Comment by JoshMcCullough** [comment]
v2.1.0: https://github.com/TeslaGov/ngx-http-auth-jwt-module/releases/tag/2.1.0

**Pull request merged**